File: sizeof.py
////OSWALDO ISAI DE LA TORRE GUERRERO////4757823////

import sys

TAMANO = 2

c = ''
x = 0
Arreglo = [0] * TAMANO
y = 0.0

print(f"\nEl tamaño en bytes de una variable char es: {sys.getsizeof(c)}")
print(f"En bits es: {sys.getsizeof(c) * 8}")
print(f"En nibbles es: {sys.getsizeof(c) * 2 // 8}")

print(f"\nEl tamaño en bytes de una variable int es: {sys.getsizeof(x)}")
print(f"En bits es: {sys.getsizeof(x) * 8}")
print(f"En nibbles es: {sys.getsizeof(x) * 2 // 8}")

print(f"\nEl tamaño en bytes de una variable arreglo es: {sys.getsizeof(Arreglo)}")
print(f"En bits es: {sys.getsizeof(Arreglo) * 8}")
print(f"En nibbles es: {sys.getsizeof(Arreglo) * 2 // 8}")

print(f"\nEl tamaño en bytes de una variable double es: {sys.getsizeof(y)}")
print(f"En bits es: {sys.getsizeof(y) * 8}")
print(f"En nibbles es: {sys.getsizeof(y) * 2 // 8}")

File: sizeof.cpp
////OSWALDO ISAI DE LA TORRE GUERRERO////4757823////

#include <stdio.h> ////bits, bytes y nibbles////

#define TAMANO 2


int main()
{
    char c;
    int x;
    int Arreglo [TAMANO];
    double y;
    
    
    printf("\n El tamaño en bytes de una variable char es: %d\n", sizeof(c));
    printf(" En bits es: %d\n",((sizeof(c))*8));
    printf(" En nibbles es: %d\n",((sizeof(c))*2));

    printf("\n El tamaño en bytes de una variable int es: %d\n", sizeof(x));
    printf(" En bits es: %d\n",((sizeof(x))*8));
    printf(" En nibbles es: %d\n",((sizeof(x))*2));
    
    printf("\n El tamaño en bytes de una variable arreglo es: %d\n", sizeof(Arreglo[TAMANO]));
    printf(" En bits es: %d\n",((sizeof(Arreglo[TAMANO]))*8));
    printf(" En nibbles es: %d\n",((sizeof(Arreglo[TAMANO]))*2));
    
    printf("\n El tamaño en bytes de una variable double es: %d\n", sizeof(y));
    printf(" En bits es: %d\n",((sizeof(y))*8));
    printf(" En nibbles es: %d\n",((sizeof(y))*2));
   return 0;
}